### ISSUE TITLE
How to handle "numbers"

### ISSUE THREAD

**Issue by kwando** [solnic/drops#33]
I tried to use a schema for a API endpoint I made that receives JSON. I this case the request originated from a JS environment like this `JSON.stringify({threshold: 1.0}) =>  "{\"threshold\":1}"`,  this is valid behavior for JSON since there is only a numeric type.

However I cant figure out how to process this data with Drops.

Naturally a schema like this will complain about `1` being an integer.
```elixir
defmodule MySchema do
  use Drops.Contract

  schema(atomize: true) do
    %{
      required(:threshold) => float()
    }
  end
end
```

After some trial and error I ended up with this, which works:
```elixir
defmodule MySchema do
  use Drops.Contract

  schema(atomize: true) do
    %{
      required(:threshold) => cast(:any, caster: __MODULE__) |> float()
    }
  end

  def cast(:any, :float, value, _) when is_number(value) do
    value * 1.0
  end
end
```

Maybe there should also be a "number" type here?

```elixir
defmodule MySchema do
  use Drops.Contract

  schema(atomize: true) do
    %{
      required(:threshold) => cast(:number) |> float()
    }
  end
end
```

**Comment by solnic** [comment]
Thanks for reporting the issue. We need a `number` type indeed :)

**Comment by solnic** [comment]
This was implemented in #41 and will be available in 0.2.0
